src/conduits/RwaMultiSwapOutputConduit.sol

```diff
@@ -0,0 +1,450 @@
+     * @param who The user address.
+     */
+    function kiss(address who) external auth {
+        bud[who] = 1;
+        emit Kiss(who);
+    }
+
+    /**
+     * @notice Remove `who` address from `pick` whitelist.
+     * @param who The user address.
+     */
+    function diss(address who) external auth {
+        if (to == who) to = address(0);
+        bud[who] = 0;
+        emit Diss(who);
+    }
+
+    /**
+     * @notice Whitelist `psm` address for `hook`.
+     * @param _psm The PSM address.
+     */
+    function clap(address _psm) external auth {
+        require(PsmAbstract(_psm).dai() == address(dai), "RwaMultiSwapOutputConduit/wrong-dai-for-psm");
+
+        // Check if GEM `decimals` is not greater then DAI decimals.
+        // We assume that DAI will always have 18 decimals
+        require(
+            GemAbstract(GemJoinAbstract(PsmAbstract(_psm).gemJoin()).gem()).decimals() <= 18,
+            "RwaMultiSwapOutputConduit/unsupported-gem-decimals"
+        );
+
+        // Give unlimited approval to PSM
+        dai.approve(_psm, type(uint256).max);
+
+        pal[_psm] = 1;
+        emit Clap(_psm);
+    }
+
+    /**
+     * @notice Remove `psm` address from `hook` whitelist.
+     * @param _psm The PSM address.
+     */
+    function slap(address _psm) external auth {
+        dai.approve(address(_psm), 0);
+        if (psm == _psm) psm = address(0);
+
+        pal[_psm] = 0;
+        emit Slap(_psm);
+    }
+
+    /*//////////////////////////////////
+               Administration
+    //////////////////////////////////*/
+
+    /**
+     * @notice Updates a contract parameter.
+     * @param what The changed parameter name. `"quitTo"`.
+     * @param data The new value of the parameter.
+     */
+    function file(bytes32 what, address data) external auth {
+        if (what == "quitTo") {
+            quitTo = data;
+        } else {
+            revert("RwaMultiSwapOutputConduit/unrecognised-param");
+        }
+
+        emit File(what, data);
+    }
+
+    /**
+     * @notice Sets `who` address as the recipient.
+     * @param who Recipient address.
+     * @dev `who` address should have been whitelisted using `kiss`.
+     */
+    function pick(address who) external {
+        require(can[msg.sender] == 1 || can[address(0)] == 1, "RwaMultiSwapOutputConduit/not-operator");
+        require(bud[who] == 1 || who == address(0), "RwaMultiSwapOutputConduit/not-bud");
+        to = who;
+        emit Pick(who);
+    }
+
+    /**
+     * @notice Sets `psm` address which will be used for swap DAI -> GEM.
+     * @param _psm PSM address.
+     * @dev `psm` address should have been whitelisted using `clap`.
+     */
+    function hook(address _psm) external {
+        require(can[msg.sender] == 1 || can[address(0)] == 1, "RwaMultiSwapOutputConduit/not-operator");
+        require(pal[_psm] == 1 || _psm == address(0), "RwaMultiSwapOutputConduit/not-pal");
+
+        psm = _psm;
+
+        emit Hook(_psm);
```

Comment:
We know this is existing from the prev version, but what is the idea to have a permissionless `pick/hook`? It can really lead to unwanted consequences due to front running. We know it is optional, but it seems a pretty broken idea excepting we are missing something that we do not know.
e.g.
- anyone can change recipient or psm to address(0) right before swapping and thus block the swaps.
- anyone can force an unaligned recipient and and psm (e.g recipient expects to get usdc, but gets usdp).
- anyone can change to an unexpected recipient (out of the allowed ones).

On the other hand, have you considered merging both functionalities in `pick` it self?
Not really sure we need to have both of them separated as the operator will need to set both of them in parallel always.

Reply:
It is permissioned, see line 314

Reply:
The comment is about the support for `can[address(0)]`.

Reply:
Make sense, it was carried from previous implementation and I think we don’t need this

Reply:
Regarding the permission system, you have a point... we probably don't want the permissionlessness here.

> On the other hand, have you considered merging both functionalities in pick it self?

We want to avoid that. We rather have people calling these contracts to perform different operations in which they can check the intermediate state than have them supply multiple arguments to the function.

There are no front-ends for these contracts, people should interact with it directly through Etherscan, so we want to make the flow as explicit as possible.

Reply:
> We want to avoid that. We rather have people calling these contracts to perform different operations in which they can check the intermediate state than have them supply multiple arguments to the function.

Fair, I'd think is more intuitive/explicit to set the whole params together but honestly it is very subjective and you know better the users, so just ignore that part of the comment.

Reply:
Looks good, this can be solved.